src/main/java/com/solace/samples/jms/snippets/HowToEnablePayloadCompression.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2021-2023 Solace Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.solace.samples.jms.snippets;
+
+import com.solacesystems.jms.SupportedProperty;
+
+import javax.jms.BytesMessage;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.jms.Topic;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.Arrays;
+import java.util.Hashtable;
+
+/**
+ * Sample of how to enable Payload Compression in JMS.
+ */
+public class HowToEnablePayloadCompression {
+    final static String SOLJMS_INITIAL_CONTEXT_FACTORY = "com.solacesystems.jndi.SolJNDIInitialContextFactory";
+    final static String CONNECTION_FACTORY_JNDI_NAME = "/JNDI/CF/GettingStarted";
+    final static String TOPIC_NAME = "T/GettingStarted/PayloadCompression";
+
+    /**
+     * Sample of how to create an {@code InitialContext} with payload compression enabled. Note: this is only when connecting to old
+     * brokers that do not have payloadCompressionLevel in the connection factory. Brokers with JMS payload compression
+     * in the connection factory WILL OVERWRITE this value.
```

Comment:
I like this comment, but let's make a bit more visible.  Add something a bit similar to the class comment above.